src/state/reducer.ts

```diff
@@ -13,6 +13,13 @@ type Writable<T> = {
   [P in keyof T]: NonNullable<T[P]>;
 }
 
+function resourceCustomizer(a: any, b: any): any {
+  if (a instanceof Resource && b instanceof Resource) {
```

Comment:
maybe just check that constructor.merge is a function? That way people can provide their own classes that still have this functionality.

Reply:
I'm not sure this is a good idea. I suspect there are (probably rare) edge cases that might cause issues that will be difficult to debug.

The way `Resource.merge()` is implemented currently for instance, 
it will only work if both objects are instances of Resource.

Seeing as the schema used for normalization can be customized, I'm not sure if entities are guaranteed to have a consistent type.

Reply:
Maybe I'm just overthinking this, and any edge cases are rare enough to not matter.

Reply:
The data structure is {entities: [Resource type key]: { [instance pk]: Instance }}}

So the invariant is that all instances at a given point in the state tree will be the same class. If that invariant is broken there is something else wrong with the system.

Reply:
Also if you use a user defined typeguard and have a type that just looks for merge as a function then you won't need the 'as typeof Resource'. (As generally considered a hack and should be avoided if possible)